doc/release-notes/rl-2505.section.md

```diff
@@ -149,6 +149,8 @@
 
 - `buildGoPackage` has been removed. Use `buildGoModule` instead. See the [Go section in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#sec-language-go) for details.
 
+- `buildGoModule` now supports a `goSum` attribute (`null` by default) to optionally provide a path to `go.sum` and enabling rebuilds when the file changes for better reproducibility
```

Comment:
I thinks this needs more explanation. Why does it provide better reproducibility?

Reply:
What I understand from it is that this attribute is now involved in the hash computation. If it changes, the resulting hash changes too.

Reply:
That's exactly what I meant. :)

Reply:
"Reproducibility" might've been a little too generic. What I had in mind wad probably something along the lines of "better dependency tracking." I'll revise the wording later today.